### ISSUE TITLE
Proposal: Change not operator priority

### ISSUE THREAD

**Issue by ceymard** [satyr/coco#85]
In python (and I believe in ruby as well), `not obj in array` -> `!(obj in array)`.

In coco, not === ! of javascript.

I find it a little counter intuitive, since when one reads "not something" in english, he understands "not the whole sentence" as opposed to "not (a single word) but the rest of the sentence is untouched".

I already fell in the trap several times writing stuff like `if not obj instanceof Array` only to realise after 30 minutes of debugging that this meant `!(obj) instanceof Array`, which would of course _never_ be true.

In short, `!something` is clearly applied to the object next to it, whereas `not something` applies to a whole expression (at least in my brain, but surely I am not alone ?).

-> `not` should be a comparison operator and have their priority instead of being a unary op.

Thoughts ?

**Comment by satyr** [comment]
- We can't do this without also changing `and`/`or` precedence:
  - Perl: http://perldoc.perl.org/perlop.html 
  - Python: http://docs.python.org/reference/expressions.html#summary
  - Ruby: http://phrogz.net/ProgrammingRuby/language.html#table_18.4
- Since CoffeeScript has chosen to [disagree](https://github.com/jashkenas/coffee-script/issues/1546), it'd still be confusing if we did this.

**Comment by satyr** [comment]
By the way, you can write `obj not instanceof Array`.
